File: app/src/main/java/fr/budgethashtag/activity/MainActivity.java
package fr.budgethashtag.activity;

import android.app.Activity;
import android.content.ContentValues;
import android.os.Bundle;
import android.widget.TextView;

import java.util.List;

import fr.budgethashtag.R;
import fr.budgethashtag.asynctask.CreateDefaultPortefeuilleIfNotExistAsyncTask;
import fr.budgethashtag.asynctask.LoadBudgetsByPortefeuilleIdAsyncTask;
import fr.budgethashtag.asynctask.LoadPortefeuilleByIdAsyncTask;
import fr.budgethashtag.asynctask.LoadTransactionsByPortefeuilleIdAsyncTask;
import fr.budgethashtag.contentprovider.PortefeuilleProvider;
import fr.budgethashtag.fragment.BudgetFragment;
import fr.budgethashtag.fragment.TransactionFragment;
import fr.budgethashtag.interfacecallbackasynctask.CreateDefaultPortefeuilleIfNotExistCallback;
import fr.budgethashtag.interfacecallbackasynctask.LoadBudgetsByPortefeuilleIdCallback;
import fr.budgethashtag.interfacecallbackasynctask.LoadPortefeuilleByIdCallback;
import fr.budgethashtag.interfacecallbackasynctask.LoadTransactionsByPortefeuilleIdCallback;

public class MainActivity extends Activity
    implements CreateDefaultPortefeuilleIfNotExistCallback,
        LoadPortefeuilleByIdCallback,
        LoadTransactionsByPortefeuilleIdCallback,
        LoadBudgetsByPortefeuilleIdCallback
{
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        new CreateDefaultPortefeuilleIfNotExistAsyncTask(this, this).execute();
    }
    @Override
    public void onCreateDefaultPortefeuilleIfNotExist(int idPortefeuille) {
        loadPortefeuilleName(idPortefeuille);
        loadTransactions(idPortefeuille);
        loadBudgets(idPortefeuille);
    }
    private void loadPortefeuilleName(int idPortefeuille) {
        new LoadPortefeuilleByIdAsyncTask(this, this, idPortefeuille).execute();
    }
    private void loadTransactions(int idPortefeuille) {
        new LoadTransactionsByPortefeuilleIdAsyncTask(this, this, idPortefeuille)
                .execute();
    }
    private void loadBudgets(int idPortefeuille) {
        new LoadBudgetsByPortefeuilleIdAsyncTask(this, this, idPortefeuille)
                .execute();
    }
    @Override
    public void onLoadPortefeuilleById(ContentValues contentValues) {
        ((TextView)findViewById(R.id.lbl_title)).setText(
                contentValues.getAsString(PortefeuilleProvider.Portefeuille.KEY_COL_LIB));
    }
    @Override
    public void onLoadTransactionsByPortefeuilleId(List<ContentValues> contentValuesList) {
        TransactionFragment transactionFragment = (TransactionFragment)getFragmentManager().findFragmentById(R.id.frg_transaction);
        transactionFragment.onLoadTransactionsByPortefeuilleId(contentValuesList);
    }

    @Override
    public void onLoadBudgetsByPortefeuilleIdCallback(List<ContentValues> contentValuesList) {
        BudgetFragment budgetFragment = (BudgetFragment)getFragmentManager().findFragmentById(R.id.frg_budget);
        budgetFragment.onLoadBudgetsByPortefeuilleIdCallback(contentValuesList);
    }

}

File: app/src/main/java/fr/budgethashtag/view/activity/MainActivity.kt
package fr.budgethashtag.view.activity

import android.content.ContentValues
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.design.widget.TabLayout
import android.support.v4.view.ViewPager
import android.support.v7.app.AppCompatActivity
import android.widget.Toolbar
import fr.budgethashtag.R
import fr.budgethashtag.contentprovider.PortefeuilleProvider
import fr.budgethashtag.databinding.ActivityMainBinding
import fr.budgethashtag.interfacecallbackasynctask.LoadBudgetsByPortefeuilleIdCallback
import fr.budgethashtag.interfacecallbackasynctask.LoadPortefeuilleByIdCallback
import fr.budgethashtag.interfacecallbackasynctask.LoadTransactionsByPortefeuilleIdCallback
import fr.budgethashtag.view.fragment.BudgetFragment
import fr.budgethashtag.view.fragment.TransactionFragment
import fr.budgethashtag.view.fragment.ViewPagerAdapter
import fr.budgethashtag.viewmodel.MainActivityViewModel


class MainActivity : AppCompatActivity(), LoadPortefeuilleByIdCallback, LoadTransactionsByPortefeuilleIdCallback, LoadBudgetsByPortefeuilleIdCallback {

    private val TAG: String = "MainActivity"
    private var viewModel = MainActivityViewModel(this)
    private lateinit var binding: ActivityMainBinding

    private lateinit var toolbar: Toolbar
    private lateinit var tabLayout: TabLayout
    private lateinit var viewPager: ViewPager

    private lateinit var transactionFragment : TransactionFragment
    private lateinit var budgetFragment : BudgetFragment

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        binding.viewModel = viewModel

        viewPager = binding.viewpager
        tabLayout = binding.tabs
        tabLayout.setupWithViewPager(viewPager)

        setSupportActionBar(binding.toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)

        budgetFragment = BudgetFragment()
        transactionFragment = TransactionFragment()
        setupViewPager(binding.viewpager)

        viewModel.onCreate(savedInstanceState)

    }

    override fun onLoadPortefeuilleById(contentValues: ContentValues) {
        binding.toolbar.title = contentValues.getAsString(PortefeuilleProvider.Portefeuille.KEY_COL_LIB)
    }

    override fun onLoadTransactionsByPortefeuilleId(contentValuesList: List<ContentValues>) {

        transactionFragment.onLoadTransactionsByPortefeuilleId(contentValuesList)
    }

    override fun onLoadBudgetsByPortefeuilleIdCallback(contentValuesList: List<ContentValues>) {
        budgetFragment.onLoadBudgetsByPortefeuilleIdCallback(contentValuesList)
    }

    private fun setupViewPager(viewPager: ViewPager) {
        val adapter = ViewPagerAdapter(supportFragmentManager)
        adapter.addFragment(budgetFragment, "Budget")
        adapter.addFragment(transactionFragment, "Transaction")
        viewPager.adapter = adapter
    }

}
